src/js/media/media.js

```diff
@@ -510,7 +511,14 @@ vjs.MediaTechController.withSourceHandlers = function(Tech){
     this.disposeSourceHandler();
     this.off('dispose', this.disposeSourceHandler);
 
-    this.currentSource_ = source;
+    // Set currentSource_ asynchronously to simulate the media element's
+    // asynchronous execution of the `resource selection algorithm`
+    setTimeout(function () {
+      if (source && source.src !== '') {
+        tech.currentSource_ = source;
+      }
+    }, 0);
```

Comment:
Very nice. If you use `this.setTimeout` it will automatically kill the timeout when the tech is disposed. Just to clean up.

Adding async always makes me worried about edge cases we're not thinking of... I can't think of anything specific though so we could pull this in and see how it goes. @videojs/core-committers any final opinions on this (setting currentSrc asynchronously or not)?